src/camera.py

```diff
@@ -0,0 +1,105 @@
+from os import getcwd
+
+from picar.SunFounder_PCA9685 import Servo
+from picar import filedb
+
+
+'''
+Camera class to control panning and tilting.
+
+Creates camera object that can be operated on using pan() and tilt().
+pan() operates [0,180] degrees. tilt() operates [0,150] degrees. These 
+functions can be calibrated using calibrate_pan() and calibrate_tilt().
+Calibration adds values to the pan() or tilt().
+'''
+class Camera:
+    PAN_CHANNEL = 1
+    TILT_CHANNEL = 2
+    MAX_PAN_DEGREE = 180
+    MAX_TILT_DEGREE = 150
+    MIN_DEGREE = 0
+    OFFSET_TO_SERVO = 0
+
+    def __init__(self, bus_number=1):
+        self._DB = filedb.fileDB(getcwd() + 
+                                 "/../SunFounder_PiCar-V/remote_control/remote_control/driver/config")
+        self._pan_offset = int(self._DB.get('pan_offset', default_value=0))
+        self._tilt_offset = int(self._DB.get('tilt_offset', default_value=0))
+        self._pan_servo = Servo.Servo(self.PAN_CHANNEL,  
+                                      bus_number=bus_number, 
+                                      offset=self.OFFSET_TO_SERVO)
+        self._tilt_servo = Servo.Servo(self.TILT_CHANNEL, 
+                                       bus_number=bus_number, 
+                                       offset=self.OFFSET_TO_SERVO)
+        self._current_pan_angle = 0
+        self._current_tilt_angle = 0
+
+    '''
+    Pans the camera.
+
+    Controls the horizontal movement of the camera in [0, 180] degrees. 
+    0 refers to the vehicle's right while 180 refers to the vehicle's left. 
+    '''
+    def pan(self, degree):
+        self._current_pan_angle = self.validate_degree(degree,
+                                                       self.MIN_DEGREE,
+                                                       self.MAX_PAN_DEGREE)
+        self._pan_servo.write(self.validate_degree(degree + self._pan_offset,
+                                                       self.MIN_DEGREE,
+                                                       self.MAX_PAN_DEGREE))
+
+    '''
+    Calibrates camera pan.
+
+    Controls offset in degrees for pan. Writes new offset value to
+    'SunFounder_PiCar-V/remote_control/remote_control/driver/config' 
+    to 'pan_offset'. 
+    '''
+    def calibrate_pan(self, degree):
+        self._DB.set('pan_offset', degree)
```

Comment:
Again, there's no validation here. I know you're validating later, but why put the system in an inconsistent state by writing wrong information to the config file.

Additionally, do you know how this config file is working? I understand that it _has_ those offsets, but _why_ does it have those offsets?

Lastly, if you're changing _DB, it's not a constant.

Reply:
Why not just validate to ensure that we can never write any value that could break the servo? Validate at every level that takes input from somewhere else that hasn't already been validated.